src/groupBy.js

```diff
@@ -48,7 +47,10 @@ var append = require('./append');
 module.exports = _curry2(_dispatchable('groupBy', _xgroupBy, function groupBy(fn, list) {
   return _reduce(function(acc, elt) {
     var key = fn(elt);
-    acc[key] = append(elt, acc[key] || (acc[key] = []));
+    if (acc[key] == null) {
```

Comment:
This check will fail for `key` values such as `'toString'` and `'valueOf'`. I'll fix this in a separate pull request.